### DISCUSSION TITLE
Consider clang_ext_vector

### DISCUSSION THREAD

**Discussion by zalo** [elalish/manifold#617] [Ideas]
Alex Evans (of Dreams-fame) writes that Clang has a special vector extension that automagically vectorizes well to SSE/AVX/NEON and has nice GLSL/HLSL-style syntax:
https://x.com/mmalex/status/1250449900514480128?s=46

Might be interesting to consider as a “drop-in” replacement for glm::vec3?

**Comment by elalish** [comment]
My understanding is that GLM has those same vectorization optimizations. Have you tried any simple performance comparisons?

**Comment by pca006132** [comment]
I feel that it would be more beneficial to work on bottlenecks such as the collider and decimator. For the collider, I think the state-of-the-art BVH are using things like multi-way trees that gives better cache performance, and the decimator is the major bottleneck for complex workload as it is sequential.
